### ISSUE TITLE
Documentation needed how to use cxxnet with any data

### ISSUE THREAD

**Issue by vkuznet** [dmlc/cxxnet#89]
Hi, I discovered cxxnet through kaggle Bowl competition and was very impressed with its performance. Now, I want to apply it to other problems beyond image recognition. Therefore it would be extremely helpful to have basic documentation how to represent data suitable as input for cxxnet.  Basically we need to feed matrix of any dimension and array of labels values. How can it be done? Thanks, Valentin.

**Comment by antinucleon** [comment]
Thanks for your words first. You may find this to be helpful https://github.com/dmlc/cxxnet/blob/master/doc/io.md#csv-iterator

**Comment by vkuznet** [comment]
Hi, it's obviously a good addition, in the code I got from antinucleon/cxxnet repository there're no doc area. So, I read short CSV part. Let me clarify a few things. I don't fully understand this sentence "In each line, the first label_width fields (default = 1, which means the label of the sample is a scalar.) are the labels for the data samples, the rest are the features for each dimension". Does it mean that I can use label_width parameter in cxxnet config file to specify position of my label column (with default value 1, i.e. first column of CSV file)? Could you please re-phrase this sentence to make it more clear. In particular user will need to know how to specifyy in cxxnet config the label column (should it be a name or position in CSV). Another useful feature can be skip labels which will tell CSV iterator to skip certain columns. In many CSV files there is an ID column which should be ignored for training purposes. So instead of regenerating CSV file it would be nice to have an option in config to ignore certain columns. Thanks, Valentin.

**Comment by winstywang** [comment]
Hi, Thanks for your comments.

1. There are a lot of “hidden features” in cxxnet now. They are experimental, and may not be stable. So we don't document it currently. After testing it more carefully, we will put it into doc. label_width is used for multilabel case. You can specify more than one labels for one sample.

2. Thanks for the advice. We will consider it in future release. You contribution is also welcome :-)

**Comment by vkuznet** [comment]
Naiyan,
how about you'll add simple description of how to use CSV. I was able to
understand it via trial procedure. For example,

The cxxnet package can read data from CSV file. The CSV file should
have the following data structure label(s)|other_columns. The number of
label columns can be controlled via label_width parameter, by default it is
set to 1, i.e. first column of CSV file is treated as labels. Also,
if file contains the header, the has_header=1 parameter should be set in
configuration (by default we assume no header in CSV file).

I found that in order to read labels the index should be properly
initialized, see pull request https://github.com/dmlc/cxxnet/pull/91

I also found that that so far has_header=1 has no effect and always read as
zero. I didn't spent much within the code, but it seems that CSV parameters
not read properly and their default values are used instead.

Thanks,
Valentin.
